pallets/gear/src/lib.rs

```diff
@@ -418,7 +419,28 @@ pub mod pallet {
             // so we don't need to include that db write.
             GasAllowanceOf::<T>::put(remaining_weight);
 
-            Self::process_queue()
+            // Ext manager creation.
+            // It will be processing messages execution results following its `JournalHandler` trait implementation.
+            // It also will handle delayed tasks following `TasksHandler`.
+            let mut ext_manager = Default::default();
+
+            // Processing regular and delayed tasks.
+            Self::process_tasks(&mut ext_manager);
```

Comment:
I would like to discuss the function in more detail. As I thought previously it's intended to bound the available weight to process such kind of tasks, for example for 5-10% of the entire block weight.

If I understand the function's logic correctly it will execute till all available gas allowance is spent in the case of too much tasks there.

Common pattern used for similar logic is to precalculate some value called (weight/gas) budget that should be 5-10% of the weight for our needs and pass the budget to the function. The procedure itslef has to check this budget before any calculation it performs. What would you say about this approach?